upf/transformers/transformer.py

```diff
@@ -122,7 +122,7 @@ def _check_and_simplify_preconditions(self, action: InstantaneousAction) -> Tupl
         #preconditions (as an And FNode)
         p = self._env.expression_manager.And(ap)
         #preconditions simplified
-        ps = self._simplifier.simplify(p)
+        ps = self._simplifier.simplify(p, self._problem)
```

Comment:
I don't think we want to use the new simplify feature for all the transformers.
Keep it optional and enable it only for the grounder for now.

Note: We do this only for InstantaneousAction: missing for the durative.

Reply:
I agree